pd/src/app.rs

```diff
@@ -155,10 +155,30 @@ impl App {
     }
 
     #[instrument(skip(self))]
-    fn deliver_tx(&mut self, tx: Bytes) -> response::DeliverTx {
+    fn deliver_tx(&mut self, hex_encoded_bytes: Bytes) -> response::DeliverTx {
         // TODO: implement (#135)
 
-        // This should accumulate data from `tx` into `self.pending_block`
+        // Transactions that cannot be deserialized should return a non-zero `DeliverTx` code.
+        let txbytes = match hex::decode(hex_encoded_bytes.as_ref()) {
```

Comment:
Can we have the `tx` be the proto-encoded bytes instead? I worry that this is going to mean that we have the blocks end up containing a bunch of hex-encoded protos.

Reply:
cf https://github.com/penumbra-zone/penumbra/pull/157/files#r750810376